### ISSUE TITLE
Feature request: Amazon SNS server-side encryption

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#6491]
Server-side encryption for Amazon SNS [recently announced](https://aws.amazon.com/about-aws/whats-new/2018/11/amazon-sns-sse/).
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/6464

**Comment by ewbankkit** [comment]
It [looks like](https://docs.aws.amazon.com/sns/latest/dg/sns-enable-encryption.html) this is done via a new SNS topic attribute `KmsMasterKeyId `.

**Comment by ewbankkit** [comment]
Proposed syntax:

```hcl
resource "aws_sns_topic" "user_updates" {
  name = "user-updates-topic"
  kms_master_key_id = "alias/aws/sns"
}
```

**Comment by bflad** [comment]
Support has been merged and will release with version 1.46.0 of the AWS provider, likely late today or tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.46.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1460-november-20-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
